File: Lib/Telerik/Examples/QuickStart/ChartView/LiveData/LiveDataModel.vb
Imports System.Text
Imports System.ComponentModel

Namespace Telerik.Examples.WinControls.ChartView.LiveData
	Friend Class LiveDataModel
		Implements INotifyPropertyChanged
'INSTANT VB NOTE: The variable data was renamed since Visual Basic does not allow class members with the same name:
		Public data_Renamed As BindingList(Of ChartBusinessObject)
'INSTANT VB NOTE: The variable data2 was renamed since Visual Basic does not allow class members with the same name:
		Public data2_Renamed As BindingList(Of ChartBusinessObject)
'INSTANT VB NOTE: The variable fps was renamed since Visual Basic does not allow class members with the same name:
		Private fps_Renamed As String
'INSTANT VB NOTE: The variable messagesPerSecond was renamed since Visual Basic does not allow class members with the same name:
		Private messagesPerSecond_Renamed As String
'INSTANT VB NOTE: The variable messagesPerMinute was renamed since Visual Basic does not allow class members with the same name:
		Private messagesPerMinute_Renamed As String
		Private tickCountSecond As Integer
		Private tickCountMinute As Integer
		Private timer As Timer
		Private lastDate As Date
		Private random As New Random()
		Private tmpDateTime As Date = Date.MinValue
		Private count As Integer = 0

		Public Sub New()
			Me.timer = New Timer()
			Me.timer.Interval = 1
			AddHandler Me.timer.Tick, AddressOf OnTimer

			Me.FillData()
			Me.FillData2()
			Me.MessagesPerSecond = Me.random.Next(800, 1800).ToString("#,#")
			Me.MessagesPerMinute = Me.random.Next(50000, 55000).ToString("#,#")
		End Sub

		Public Property FPS() As String
			Get
				Return Me.fps_Renamed
			End Get
			Set(ByVal value As String)
				If Me.fps_Renamed <> value Then
					Me.fps_Renamed = value
					Me.OnPropertyChanged("FPS")
				End If
			End Set
		End Property

		Public Property MessagesPerSecond() As String
			Get
				Return Me.messagesPerSecond_Renamed
			End Get
			Set(ByVal value As String)
				If Me.messagesPerSecond_Renamed <> value Then
					Me.messagesPerSecond_Renamed = value
					Me.OnPropertyChanged("MessagesPerSecond")
				End If
			End Set
		End Property

		Public Property MessagesPerMinute() As String
			Get
				Return Me.messagesPerMinute_Renamed
			End Get
			Set(ByVal value As String)
				If Me.messagesPerMinute_Renamed <> value Then
					Me.messagesPerMinute_Renamed = value
					Me.OnPropertyChanged("MessagesPerMinute")
				End If
			End Set
		End Property

		Public Property Data() As BindingList(Of ChartBusinessObject)
			Get
				Return Me.data_Renamed
			End Get
			Set(ByVal value As BindingList(Of ChartBusinessObject))
				If Me.data_Renamed IsNot value Then
					Me.data_Renamed = value
					Me.OnPropertyChanged("Data")
				End If
			End Set
		End Property

		Public Property Data2() As BindingList(Of ChartBusinessObject)
			Get
				Return Me.data2_Renamed
			End Get
			Set(ByVal value As BindingList(Of ChartBusinessObject))
				If Me.data2_Renamed IsNot value Then
					Me.data2_Renamed = value
					Me.OnPropertyChanged("Data2")
				End If
			End Set
		End Property

		Public Sub StartTimer()
			Me.timer.Start()
		End Sub

		Public Sub StopTimer()
			Me.timer.Stop()
		End Sub

		Public Sub UpdateTimer(ByVal interval As Double)
			Me.timer.Interval = CInt(Fix(interval))
		End Sub

		Private Sub FillData()
			Dim collection As New BindingList(Of ChartBusinessObject)()
			Me.lastDate = Date.Now

			For i As Integer = 0 To 49
				Me.lastDate = Me.lastDate.AddMilliseconds(200)
				collection.Add(Me.CreateBusinessObject())
			Next i

			Me.Data = collection
		End Sub

		Private Sub FillData2()
			Dim collection As New BindingList(Of ChartBusinessObject)()
			Dim [date] As New Date(Date.Now.Year, Date.Now.Month, Date.Now.Day)

			For i As Integer = 0 To 23
				collection.Add(Me.CreateBusinessObject2([date].AddHours(i)))
			Next i

			Me.Data2 = collection
		End Sub

		Private Sub OnTimer(ByVal sender As Object, ByVal e As EventArgs)
			Dim now As Date = Date.Now
			If (now.Subtract(tmpDateTime)).TotalSeconds > 1 Then
				tmpDateTime = now
				If count <> 0 Then
					Me.FPS = (count \ 1).ToString()
				End If

				count = 0
			Else
				count += 1
			End If

			Me.lastDate = Me.lastDate.AddMilliseconds(200)
			Me.Data.RemoveAt(0)
			Me.Data.Add(Me.CreateBusinessObject())
			Me.UpdateMetrics()
		End Sub

		Private Sub UpdateMetrics()
			Me.tickCountSecond += 1
			Me.tickCountMinute += 1

			If Me.tickCountSecond = 5 Then
				Me.tickCountSecond = 0
				Me.MessagesPerSecond = Me.random.Next(800, 1800).ToString("#,#")
			End If

			If Me.tickCountMinute = 100 Then
				Me.tickCountMinute = 0
				Me.MessagesPerMinute = Me.random.Next(50000, 55000).ToString("#,#")
			End If
		End Sub

		Private Function CreateBusinessObject() As ChartBusinessObject
			Dim obj As New ChartBusinessObject()

			obj.Value = Me.random.Next(800, 1800)
			obj.Category = Me.lastDate

			Return obj
		End Function

		Private Function CreateBusinessObject2(ByVal [date] As Date) As ChartBusinessObject
			Dim obj As New ChartBusinessObject()

			obj.Value = Me.random.Next(3300, 3800)
			obj.Category = [date]

			Return obj
		End Function

		Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

		Protected Overridable Sub OnPropertyChanged(ByVal propertyName As String)
			RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(propertyName))
		End Sub

	End Class
End Namespace

File: Lib/Telerik/Examples/QuickStart/ChartView/LiveData/LiveDataModel.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.ComponentModel;
using System.Windows.Forms;

namespace Telerik.Examples.WinControls.ChartView.LiveData
{
    class LiveDataModel : INotifyPropertyChanged
    {
        public BindingList<ChartBusinessObject> data;
        public BindingList<ChartBusinessObject> data2;
        private string fps;
        private string messagesPerSecond;
        private string messagesPerMinute;
        private int tickCountSecond;
        private int tickCountMinute;
        private Timer timer;
        private DateTime lastDate;
        private Random random = new Random();
        private DateTime tmpDateTime = DateTime.MinValue;
        private int count = 0;

        public LiveDataModel()
        {
            this.timer = new Timer();
            this.timer.Interval = 1;
            this.timer.Tick += this.OnTimer;

            this.FillData();
            this.FillData2();
            this.MessagesPerSecond = this.random.Next(800, 1800).ToString("#,#");
            this.MessagesPerMinute = this.random.Next(50000, 55000).ToString("#,#");
        }

        public string FPS
        {
            get
            {
                return this.fps;
            }
            set
            {
                if (this.fps != value)
                {
                    this.fps = value;
                    this.OnPropertyChanged("FPS");
                }
            }
        }

        public string MessagesPerSecond
        {
            get
            {
                return this.messagesPerSecond;
            }
            set
            {
                if (this.messagesPerSecond != value)
                {
                    this.messagesPerSecond = value;
                    this.OnPropertyChanged("MessagesPerSecond");
                }
            }
        }

        public string MessagesPerMinute
        {
            get
            {
                return this.messagesPerMinute;
            }
            set
            {
                if (this.messagesPerMinute != value)
                {
                    this.messagesPerMinute = value;
                    this.OnPropertyChanged("MessagesPerMinute");
                }
            }
        }

        public BindingList<ChartBusinessObject> Data
        {
            get
            {
                return this.data;
            }
            set
            {
                if (this.data != value)
                {
                    this.data = value;
                    this.OnPropertyChanged("Data");
                }
            }
        }

        public BindingList<ChartBusinessObject> Data2
        {
            get
            {
                return this.data2;
            }
            set
            {
                if (this.data2 != value)
                {
                    this.data2 = value;
                    this.OnPropertyChanged("Data2");
                }
            }
        }

        public void StartTimer()
        {
            this.timer.Start();
        }

        public void StopTimer()
        {
            this.timer.Stop();
        }

        public void UpdateTimer(double interval)
        {
            this.timer.Interval = (int)interval;
        }

        private void FillData()
        {
            BindingList<ChartBusinessObject> collection = new BindingList<ChartBusinessObject>();
            this.lastDate = DateTime.Now;

            for (int i = 0; i < 50; i++)
            {
                this.lastDate = this.lastDate.AddMilliseconds(200);
                collection.Add(this.CreateBusinessObject());
            }

            this.Data = collection;
        }

        private void FillData2()
        {
            BindingList<ChartBusinessObject> collection = new BindingList<ChartBusinessObject>();
            DateTime date = new DateTime(DateTime.Now.Year, DateTime.Now.Month, DateTime.Now.Day);

            for (int i = 0; i < 24; i++)
                collection.Add(this.CreateBusinessObject2(date.AddHours(i)));

            this.Data2 = collection;
        }

        private void OnTimer(object sender, EventArgs e)
        {
            DateTime now = DateTime.Now;
            if ((now - tmpDateTime).TotalSeconds > 1)
            {
                tmpDateTime = now;
                if (count != 0)
                {
                    this.FPS = (count / 1).ToString();
                }

                count = 0;
            }
            else
            {
                count++;
            }

            this.lastDate = this.lastDate.AddMilliseconds(200);
            this.Data.RemoveAt(0);
            this.Data.Add(this.CreateBusinessObject());
            this.UpdateMetrics();
        }

        private void UpdateMetrics()
        {
            this.tickCountSecond++;
            this.tickCountMinute++;

            if (this.tickCountSecond == 5)
            {
                this.tickCountSecond = 0;
                this.MessagesPerSecond = this.random.Next(800, 1800).ToString("#,#");
            }

            if (this.tickCountMinute == 100)
            {
                this.tickCountMinute = 0;
                this.MessagesPerMinute = this.random.Next(50000, 55000).ToString("#,#");
            }
        }

        private ChartBusinessObject CreateBusinessObject()
        {
            ChartBusinessObject obj = new ChartBusinessObject();

            obj.Value = this.random.Next(800, 1800);
            obj.Category = this.lastDate;

            return obj;
        }

        private ChartBusinessObject CreateBusinessObject2(DateTime date)
        {
            ChartBusinessObject obj = new ChartBusinessObject();

            obj.Value = this.random.Next(3300, 3800);
            obj.Category = date;

            return obj;
        }

        public event PropertyChangedEventHandler PropertyChanged;

        protected virtual void OnPropertyChanged(string propertyName)
        {
            if (this.PropertyChanged != null)
            {
                this.PropertyChanged(this, new PropertyChangedEventArgs(propertyName));
            }
        }

    }
}
